src/sigverify.rs

```diff
@@ -414,9 +424,9 @@ mod tests {
         let (sig_len, sig_start, msg_start_offset, pubkey_offset) =
             sigverify::get_packet_offsets(&packet, 0);
         assert_eq!(sig_len, 1);
-        assert_eq!(sig_start, 8);
-        assert_eq!(msg_start_offset, 72);
-        assert_eq!(pubkey_offset, 80);
+        assert_eq!(sig_start, 9);
```

Comment:
I would have guessed that'd be 1, not 9. What's in those first 8 bytes?

Reply:
After we serialize the transaction into bytes, we end up calling serialize_bytes that adds up the 8 bytes to the size.

Reply:
why?

Reply:
Transaction structure requires serialize / deserialize traits to be implemented as otherwise it would require to hand roll the serialize / deserialize for all structures that include Transaction (there seems to be a number of places where this gets included) as serde derive could not be used and would complain that these traits are not implemented.

The serialize operation does not allow us to currently override with our own Serializer and hence we manually serialize the individual members of the Transaction and then invoke the serializer once on this stream of serialized bytes which ends up adding the 8 bytes to indicate the size.